File: lib/src/main/java/com/orientsec/easysocket/session/State.java
package com.orientsec.easysocket.session;

/**
 * Represents the connection state of a session.
 *
 * <p>
 * IDLE: The idle state. Transition: IDLE -> STARTING.
 * </p>
 *
 * <p>
 * STARTING: The starting state.
 * 1. Transition: STARTING -> DETACHED (connection failed).
 * 2. Transition: STARTING -> CONNECTED (connection succeeded).
 * </p>
 *
 * <p>
 * CONNECTED: The connected state.
 * 1. Transition: CONNECTED -> AVAILABLE (initialization succeeded).
 * 2. Transition: CONNECTED -> DETACHED (initialization failed).
 * </p>
 *
 * <p>
 * AVAILABLE: The available state. Transition: AVAILABLE -> DETACHED (connection lost).
 * </p>
 *
 * <p>
 * DETACHED: The detached state, indicating the session is no longer active.
 * </p>
 */
enum State {
    IDLE, STARTING, CONNECTED, AVAILABLE, DETACHED
}

File: library/core/src/commonMain/kotlin/com/orientsec/easysocket/session/State.kt
package com.orientsec.easysocket.session

/**
 * 会话的连接状态枚举。
 *
 * 状态流转图：
 * ```
 * IDLE -> STARTING -> CONNECTED -> AVAILABLE -> DETACHED
 *                  \-> DETACHED  \-> DETACHED
 * ```
 *
 * 各状态说明：
 * - [IDLE]: 初始空闲状态，会话刚创建尚未开始连接
 * - [STARTING]: 正在连接中，Socket 连接正在建立
 * - [CONNECTED]: Socket 连接已建立，等待会话初始化（如登录）
 * - [AVAILABLE]: 会话可用，所有请求可以正常发送
 * - [DETACHED]: 已断开，会话不再活跃
 */
enum class State {
    /** 空闲状态，会话尚未启动 */
    IDLE,
    /** 正在连接中 */
    STARTING,
    /** 已连接，等待初始化完成 */
    CONNECTED,
    /** 可用状态，可以发送请求 */
    AVAILABLE,
    /** 已断开，会话结束 */
    DETACHED
}